### PR TITLE
test: migrate spec-test-utils and spec tests to vitest

### PR DESCRIPTION
**Pull request by nazarhussain**
**Motivation**

Consolidate the testing frameworks and migrate to vitest.

**Description**

Migrate `spec-test-utils` and spec tests to vitest.

**Steps to test or reproduce**

- Run all tests

### CODE REVIEW

File: packages/spec-test-util/package.json [hunk 1]
```diff
@@ -9,7 +9,14 @@
...
   "type": "module",
-  "exports": "./lib/index.js",
+  "exports": {
+    ".": {
+      "import": "./lib/index.js"
+    },
+    "./download": {
+      "import": "./lib/downloadTests.js"
+    }
+  },
```
Comment (nazarhussain) [inline packages/spec-test-util/package.json]: Any code which imports `vitest` should be run via `vitest` binary. For download script we need to run it directly with `nodejs` so why have to extract that part as separate export.

File: packages/spec-test-util/package.json [hunk 2]
```diff
@@ -26,12 +33,13 @@
     "build": "tsc -p tsconfig.build.json",
     "build:release": "yarn clean && yarn build",
     "build:watch": "yarn run build --watch",
-    "check-build": "node -e \"(async function() { await import('./lib/index.js') })()\"",
+    "check-build": "node -e \"(async function() { await import('./lib/downloadTests.js') })()\"",
```
Comment (nazarhussain) [inline packages/spec-test-util/package.json:45]: As mentioned in above comment.

File: packages/beacon-node/test/spec/presets/fork_choice.test.ts [hunk 3]
```diff
@@ -112,6 +112,9 @@ const forkChoiceTest =
           }
         );
 
+        // The handler of `ChainEvent.forkChoiceFinalized` access `db.block` and raise error if not found.
+        chain.emitter.removeAllListeners(ChainEvent.forkChoiceFinalized);
```
Comment (nazarhussain) [inline packages/beacon-node/test/spec/presets/fork_choice.test.ts:116]: This handler is throwing an error `block not found`, which is `unhandled rejection` even in previous implementation but never been caught by mocha.

File: packages/validator/test/spec/spec.test.ts [hunk 4]
```diff
@@ -22,42 +22,44 @@ describe("slashing-protection-interchange-tests", () => {
...
   for (const testCase of testCases) {
     describe(testCase.name, () => {
       for (const step of testCase.steps) {
-        beforeEach(async () => {
-          await db.clear();
-        });
+        // If there is no `it` block then we should skip to avoid running `beforeEach` hooks
+        if (step.blocks.length === 0 && step.attestations.length === 0) {
+          continue;
+        }
```
Comment (nazarhussain) [inline packages/validator/test/spec/spec.test.ts:41]: This is a very interesting case to show difference of both test runners. 

1. Mocha don't run the `beforeEach` which is defined within the loop without `it`.
2. Vitest consider all `it` blocks and `beforeEach` blocks as children of the `describe` and run two `beforeEach` for single `it` block.

**Review by wemeetagain — APPROVED** [review]

**Comment by wemeetagain** [comment]
:tada: This PR is included in [v1.15.0](https://github.com/ChainSafe/lodestar/releases/tag/v1.15.0) :tada:

**Pull request merged**